src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -137,10 +139,9 @@ internal sealed class SolutionRestoreWorker : SolutionEventsListener, ISolutionR
             // Needed when OnAfterBackgroundSolutionLoadComplete fires before
             // Advise has been called.
             // IsSolutionFullyLoaded may be expensive so make sure it's called only once.
-            if (!_solutionLoadedTcs.Task.IsCompleted
-                && SolutionManager.IsSolutionFullyLoaded)
+            if (SolutionManager.IsSolutionFullyLoaded)
```

Comment:
Your perf optimization to avoid calling this property is fine, if you want to add `_solutionLoadedEvent.IsSet &&` back into this expression.